### ISSUE TITLE
`cached` decorator creates new backend per decorated function

### ISSUE THREAD

**Issue by popravich** [argaen/aiocache#50]
```python
def cached(cache=None,...):
    cache = get_cache(cache=cache, ...)
    # ...
    def cached_decorator(fn):
        # ...
    return cached_decorator
```
Basically, `get_cache` returns new instance of backend for every call.
This mean that every decorated function will have its own instance of backend
and that every such function will create at least one connection to some
remote resource (redis, memcached, etc) (only if its not MemoryBackend)
and this might become a problem.

**Comment by argaen** [comment]
Good catch, I have to think a way of how to reuse the same backend if its always the same but create a new one if the user sends the `endpoint` or other variables that affect the underlying backend.

Thanks for reporting!

**Comment by argaen** [comment]
Im moving the discussion to #45 since its related with the `set_defaults` and `get_cache` behavior.

**Comment by argaen** [comment]
Fixed by #77
